pkgs/development/tools/rust/cargo-leptos/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, rustPlatform , fetchFromGitHub, Security, lib, pkg-config, openssl }:
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-leptos";
+  version = "0.1.8";
+  buildFeatures = [ "no_downloads" ]; # cargo-leptos will try to download Ruby and other things without this feature
+
+  src = fetchFromGitHub {
+    owner = "leptos-rs";
+    repo = pname;
+    rev = version;
+    hash = "sha256-z4AqxvKu9E8GGMj6jNUAAWeqoE/j+6NoAEZWeNZ+1BA=";
+  };
+
+  cargoSha256 = "sha256-w/9W4DXbh4G5DZ8IGUz4nN3LEjHhL7HgybHqODMFzHw=";
+
+  nativeBuildInputs = [ pkg-config openssl ];
+
+  buildInputs = [ openssl pkg-config ] ++ lib.optionals stdenv.isDarwin [
+    Security
```

Comment:
For the build to succeed on Darwin, `CoreServices` are needed as a `buildInput` as well. Coincidentally, they are the only package dependencies needed, `openssl` and `pkg-config` can be wrapped in `lib.optionals. stdenv.isLinux`. At least on `aarch64-darwin` that I tested it on.